services/controlplane/features/client.go

```diff
@@ -0,0 +1,123 @@
+package features
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/services/controlplane/identity"
+)
+
+var (
+	DefaultTimeout = 30 * time.Second
+	MaxRetries     = uint64(3)
+)
+
+type OptFn func(c *Client)
+
+func WithURL(url string) OptFn {
+	return func(c *Client) {
+		c.url = url
+	}
+}
+
+func WithHTTPClient(httpClient *http.Client) OptFn {
+	return func(c *Client) {
+		c.client = httpClient
+	}
+}
+
+func WithTimeout(timeout time.Duration) OptFn {
+	return func(c *Client) {
+		c.client.Timeout = timeout
+	}
+}
+
+type Client struct {
+	client   *http.Client
+	url      string
+	identity identity.Identifier
+}
+
+type payload struct {
+	Components []component `json:"components"`
+}
+
+type component struct {
+	Name     string   `json:"name"`
+	Features []string `json:"features"`
+}
+
+func New(identity identity.Identifier, fns ...OptFn) *Client {
+	c := &Client{
+		client: &http.Client{
+			Timeout: DefaultTimeout,
+		},
+		url:      config.GetEnv("CONFIG_BACKEND_URL", "https://api.rudderlabs.com"),
```

Comment:
shouldn't the `url` be externally configurable as well?

Reply:
But it is right? 
 https://github.com/rudderlabs/rudder-server/pull/2377/files/503743e24501d634a0f9b17e800d77f2af357282#diff-df67db07b56f23e5559aced20214af1134daefe6eec3bca096ef3807ed04e91cR25

Reply:
oops!